### PR TITLE
Retrieve the table handle only when dealing with a system table

### PR DESCRIPTION
**Pull request by findinpath**
## Description

This is a small usability improvement which aims to provide a more meaningful error message to the user in case that the user asks for an invalid metadata table corresponding to an existing Iceberg table.

```
assertQueryFailure(() -> onTrino().executeQuery("TABLE iceberg.default.\"" + tableName + "$invalid\""))
                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): Table '" + icebergTableName + "$invalid' redirected to '" + hiveTableName + "$invalid', but the target table '" + hiveTableName + "$invalid' does not exist");
```

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -434,15 +434,11 @@ public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTabl
 
     private Optional<SystemTable> getRawSystemTable(ConnectorSession session, SchemaTableName tableName)
     {
-        IcebergTableName name = IcebergTableName.from(tableName.getTableName());
-        if (name.getTableType() == DATA) {
-            return Optional.empty();
-        }
-
+        String name = IcebergTableName.tableNameFrom(tableName.getTableName());
```
Comment (findinpath) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:437]: The table handle should be retrieved only when dealing with non-DATA system tables.

**Comment by findinpath** [comment]
Yes, at the moment is not reviewable. I made a false supposition while writing it.

**Comment by findinpath** [comment]
CI Hit https://github.com/trinodb/trino/issues/5427

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 2]
```diff
@@ -345,11 +345,11 @@ public IcebergTableHandle getTableHandle(
             throw new TrinoException(NOT_SUPPORTED, "Read table with start version is not supported");
```
Comment (findinpath) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:345]: FWIW the logic is now the same as the logic employed in `DeltaLakeMetadata`:  

https://github.com/trinodb/trino/blob/92e5d111c0dda3d5a9d24a34f975e78724e2e098/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L2506

**Comment by findinpath** [comment]
@ebyhr  CPTAL at this PR ? It is pretty similar to what has been done for the `$history` metadata table in Delta.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableName.java [hunk 3]
```diff
@@ -81,4 +83,53 @@ public static IcebergTableName from(String name)
...
+        }
+
+        return match.group("table");
+    }
+
+    public static Optional<TableType> tableTypeFrom(String name)
+    {
+        Matcher match = TABLE_PATTERN.matcher(name);
+        if (!match.matches()) {
+            throw new TrinoException(NOT_SUPPORTED, "Invalid Iceberg Lake table name: " + name);
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableName.java:101]:
```suggestion
            throw new TrinoException(NOT_SUPPORTED, "Invalid Iceberg table name: " + name);
```

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableName.java [hunk 4]
```diff
@@ -81,4 +83,53 @@ public static IcebergTableName from(String name)
...
+        }
+        String typeString = match.group("type");
+        if (typeString == null) {
+            return Optional.of(DATA);
+        }
+        try {
+            return Optional.of(TableType.valueOf(typeString.toUpperCase(Locale.ENGLISH)));
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableName.java:108]: nit: Static import `ENGLISH`. Also, can we change existing `Locale.ROOT` to `ENGLISH`?

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
CI hit #15187

**Pull request merged**